modules/nf-core/custom/tabulartogseagct/main.nf

```diff
@@ -0,0 +1,33 @@
+process CUSTOM_TABULARTOGSEAGCT {
+    tag "$meta.id"
+    label 'process_single'
+
+    input:
+    tuple val(meta), path(tabular)
+
+    output:
+    tuple val(meta), path("*.gct"), emit: gct
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def separator = tabular.getName().endsWith(".csv") ? ',' : '\\t'
+    """
+    n_columns=\$(head -n 1 $tabular | tr "$separator" "\\n" | wc -l)
```

Comment:
Also this should be possible in Groovy code inside your pipeline, but would probably take a bit more code. For me it's okay to add this to `custom`, especially if it could be used by other people to work with the GSEA modules